src/flp.rs

```diff
@@ -490,6 +494,46 @@ pub trait Type: Sized + Eq + Clone + Debug {
 
         Ok(true)
     }
+
+    /// Check whether `input` and `joint_rand` have the length expected by `self`,
+    /// return [`FlpError::Valid`] otherwise.
+    fn valid_call_check(
+        &self,
+        input: &[Self::Field],
+        joint_rand: &[Self::Field],
+    ) -> Result<(), FlpError> {
+        if input.len() != self.input_len() {
```

Comment:
nit: You could call `truncate_call_check()` instead in order to de-duplicate code.

Reply:
Yes, though the problem would be that `truncate_call_check()` returns an `FlpError::Truncate` while `valid_call_check()` returns an `FlpError::Valid`.

We could do either:
- Choose another error type which we throw in both functions.
- Call `truncate_call_check`, then do `map_err` to change the error type.

Reply:
Ack, good catch. You can leave as is.